src/components/Pagination.tsx

```diff
@@ -44,10 +47,44 @@ export function Pagination<Entry>({
     lastPageNumber - 1 !== numberOfButtons;
   const buttonNumbers = makeButtonNumbers();
 
+  const getPageNumberOfItem = useCallback(
```

Comment:
I wrapped up these functions that get used in dependency arrays in `useCallback`. Usually this is not necessary, but when a function is used in a dependency array without `useCallback` it will cause the effect to fire on every render, because a function is a different object on each invocation of the component unless memoized like so.